### PR TITLE
Refactor actions-plan-preview

### PR DESCRIPTION
**Pull request by gkuga**
**What this PR does / why we need it**:

I want to support push events in actions-plan-preview. Before adding that functionality, I refactored it to make it testable.

* Refactor parseGitHubEvent
* Add tests
  * testdata is from [GitHub webhook payload sample](https://docs.github.com/ja/developers/webhooks-and-events/webhooks/webhook-events-and-payloads)
* Fix failing test

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: dockers/actions-plan-preview/github.go [hunk 1]
```diff
@@ -115,15 +116,15 @@ func parseGitHubEvent(ctx context.Context, client *github.Client) (*githubEvent,
 	}
 }
 
-func sendComment(ctx context.Context, client *github.Client, owner, repo string, prNum int, body string) (*github.IssueComment, error) {
-	c, _, err := client.Issues.CreateComment(ctx, owner, repo, prNum, &github.IssueComment{
+func sendComment(ctx context.Context, issuesService IssuesService, owner, repo string, prNum int, body string) (*github.IssueComment, error) {
+	c, _, err := issuesService.CreateComment(ctx, owner, repo, prNum, &github.IssueComment{
```
Comment (nghialv) [inline dockers/actions-plan-preview/github.go:120]:
```suggestion
func sendComment(ctx context.Context, svc IssuesService, owner, repo string, prNum int, body string) (*github.IssueComment, error) {
	c, _, err := svc.CreateComment(ctx, owner, repo, prNum, &github.IssueComment{
```

File: dockers/actions-plan-preview/github.go [hunk 2]
```diff
@@ -40,30 +39,32 @@ type githubEvent struct {
...
-
-	eventPath := os.Getenv("GITHUB_EVENT_PATH")
-	payload, err := os.ReadFile(eventPath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read event payload: %v", err)
-	}
-
+func parseGitHubEvent(
+	ctx context.Context,
+	prsService PullRequestsService,
```
Comment (nghialv) [inline dockers/actions-plan-preview/github.go:64]:
```suggestion
	pullSvc PullRequestsService,
```

File: dockers/actions-plan-preview/github.go [hunk 3]
```diff
@@ -90,7 +91,7 @@ func parseGitHubEvent(ctx context.Context, client *github.Client) (*githubEvent,
 			repo  = e.Repo.GetName()
 			prNum = e.Issue.GetNumber()
 		)
-		pr, err := getPullRequest(ctx, client, owner, repo, prNum)
+		pr, err := getPullRequest(ctx, prsService, owner, repo, prNum)
```
Comment (nghialv) [inline dockers/actions-plan-preview/github.go:94]:
```suggestion
		pr, err := getPullRequest(ctx, pullSvc, owner, repo, prNum)
```

File: dockers/actions-plan-preview/main_test.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	"embed"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+//go:embed testdata/*
+var testdata embed.FS
+
+func readFile(t *testing.T, name string) []byte {
```
Comment (nghialv) [inline dockers/actions-plan-preview/main_test.go:27]: nit: `readTestdataFile`

**Comment by nghialv** [comment]
@gkuga Thanks for your pull request.

Looks nice. I have just left some nits.
PTAL.

File: dockers/actions-plan-preview/main.go [hunk 5]
```diff
@@ -183,3 +202,13 @@ func parseArgs(args []string) (arguments, error) {
...
 }
+
+func isSupportedGitHubEvent(event string) bool {
+	switch event {
+	case pullRequestEventName:
+		return true
+	case commentEventName:
+		return true
+	}
+	return false
```
Comment (knanao) [inline dockers/actions-plan-preview/main.go:213]: nits
```suggestion
	return event == pullRequestEventName || event == commentEventNam
```

File: dockers/actions-plan-preview/main.go [hunk 6]
```diff
@@ -42,6 +44,21 @@ func main() {
...
 	log.Println("Successfully parsed arguments")
 
+	eventName := os.Getenv("GITHUB_EVENT_NAME")
+	if !isSupportedGitHubEvent(eventName) {
+		log.Fatal(fmt.Errorf(
+			"unexpected event %s, only %q and %q event are supported",
+			eventName,
+			pullRequestEventName,
+			commentEventName,
+		))
```
Comment (knanao) [inline dockers/actions-plan-preview/main.go:54]: Can we return early if we get unexpected event?

**Comment by gkuga** [comment]
I addressed all the review comments, please review it again:pray:

**Comment by nghialv** [comment]
Great improvement. Thank you.
/lgtm

**Comment by knanao** [comment]
Here you go!
/approve

**Pull request merged**